pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -292,6 +321,187 @@ func (a Adapter) pushLocal(path string, files []string, delFiles []string, isFor
 	return nil
 }
 
+// Push syncs source code from the user's disk to the component
+func (a Adapter) execDevfile(pushDevfileCommands []versionsCommon.DevfileCommand, componentExists, show bool) error {
```

Comment:
I'm wondering...

We'll need the exec code to be pretty consistent between local (Docker) and Kube. I'm wondering if we should move this out to a new package, like we did for sync (let's call it `exec`). Then define an interface (let's call it ExecClient) that has one function: `ExecCmdInContainer`, and both the `kclient` and `lclient` would implement it.

It's up to you, but moving this out to its own package now, makes our lives easier for Docker support.

Reply:
Sounds good, let me do it like the way you did for sync, to be consistent